src/http/sr-modules-validators/sr-modules-validators.service.ts

```diff
@@ -5,164 +5,141 @@ import {
-    // We need EL meta always be actual
-    if (elMeta.blockNumber < clMeta.blockNumber) {
-      this.logger.warn('Last Execution Layer block number in our database older than last Consensus Layer');
-      // add metric or alert on breaking el > cl condition
-      // TODO: what answer will be better here?
-      // TODO: describe in doc
-      throw new InternalServerErrorException(
-        'Last Execution Layer block number in our database older than last Consensus Layer',
-      );
-    }
-
-    return { validators, meta: clMeta };
+    return {
+      data,
+      meta: {
+        clBlockSnapshot: clBlockSnapshot,
+      },
+    };
   }
 
   private createExitValidatorList(validators: Validator[]): ExitValidator[] {
     return validators.map((v) => ({ validatorIndex: v.index, key: v.pubkey }));
   }
 
-  private createExitPresignMessageList(validators: Validator[], clMeta: ConsensusMeta): ExitPresignMessage[] {
+  private createExitPresignMessageList(validators: Validator[], clMeta: CLBlockSnapshot): ExitPresignMessage[] {
     return validators.map((v) => ({ validator_index: String(v.index), epoch: String(clMeta.epoch) }));
   }
 
-  private disabledRegistry() {
+  private isRegistryDisabled() {
     return !this.configService.get('VALIDATOR_REGISTRY_ENABLE');
   }
+
+  private async getOperatorOldestValidators(
+    moduleId: string,
+    operatorIndex: number,
+    filters: ValidatorsQuery,
+  ): Promise<{ validators: Validator[]; clBlockSnapshot: CLBlockSnapshot }> {
+    const { validators, meta } = await this.entityManager.transactional(
+      async () => {
+        const { module, elBlockSnapshot } = await this.stakingRouterService.getStakingModuleAndMeta(moduleId);
+
+        // read from config name of module that implement functions to fetch and store keys for type
+        // TODO: check what will happen if implementation is not a provider of StakingRouterModule
+        const moduleInstance = this.stakingRouterService.getStakingRouterModuleImpl(module.type);
+        const keys = await moduleInstance.getKeys(module.stakingModuleAddress, { operatorIndex }, ['key']);
+
+        const pubkeys = keys.map((pubkey) => pubkey.key);
+        const percent =
+          filters?.max_amount == undefined && filters?.percent == undefined ? DEFAULT_EXIT_PERCENT : filters?.percent;
+
+        const result = await this.validatorsService.getOldestValidators({
+          pubkeys,
+          statuses: VALIDATORS_STATUSES_FOR_EXIT,
+          max_amount: filters?.max_amount,
+          percent: percent,
+        });
+
+        if (!result) {
+          // if result of this method is null it means Validators Registry is disabled
+          throw new InternalServerErrorException(VALIDATORS_REGISTRY_DISABLED_ERROR);
+        }
+
+        const { validators, meta: clMeta } = result;
+
+        // check if clMeta is not null
+        // if it is null, it means keys db is empty and Updating Validators Job is not finished yet
+        if (!clMeta) {
```

Comment:
Maybe this should be checked in middleware and removed everywhere in views?

Reply:
will check how to simplify it in the next iteration